binder/src/main/java/io/grpc/binder/SecurityPolicies.java

```diff
@@ -187,4 +188,47 @@ private static boolean checkPackageSignature(
     }
     return false;
   }
+
+  /**
+   * Creates a {@link SecurityPolicy} which checks if the caller has all of the given permissions
+   * from {@code permissions}.
+   *
+   * @param permissions all permissions that the calling package needs to have
+   * @throws NullPointerException if any of the inputs are {@code null}
+   * @throws IllegalArgumentException if {@code permissions} is empty
+   */
+  public static SecurityPolicy hasPermissions(
+      PackageManager packageManager, ImmutableSet<String> permissions) {
+    Preconditions.checkNotNull(packageManager, "packageManager");
+    Preconditions.checkNotNull(permissions, "permissions");
+    Preconditions.checkArgument(!permissions.isEmpty(), "permissions");
+    return new SecurityPolicy() {
+      @Override
+      public Status checkAuthorization(int uid) {
+        return checkPermissions(uid, packageManager, permissions);
+      }
+    };
+  }
+
+  private static Status checkPermissions(
+      int uid, PackageManager packageManager, ImmutableSet<String> permissions) {
+    String[] packages = packageManager.getPackagesForUid(uid);
+    if (packages == null) {
```

Comment:
Check the array size >0 as well? Otherwise we'll skip the check. Even if the Android API claims to return null, it feels reasonable to be paranoid in this case.

Reply:
should be good now